jdaviz/configs/default/plugins/markers/markers.py

```diff
@@ -0,0 +1,150 @@
+import numpy as np
+from traitlets import observe
+
+from jdaviz.core.events import ViewerAddedMessage
+from jdaviz.core.marks import MarkersMark
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import PluginTemplateMixin, ViewerSelectMixin, TableMixin
+from jdaviz.core.user_api import PluginUserApi
+
+__all__ = ['Markers']
+
+_default_table_values = {'spectral_axis': np.nan,
+                         'spectral_axis:unit': '',
+                         'slice': np.nan,
+                         'pixel': (np.nan, np.nan),
+                         'pixel:unreliable': None,
+                         'world': (np.nan, np.nan),
+                         'world:unreliable': None,
+                         'value': np.nan,
+                         'value:unit': '',
+                         'value:unreliable': None,
+                         'index': np.nan}
+
+
+@tray_registry('g-markers', label="Markers")
+class Markers(PluginTemplateMixin, ViewerSelectMixin, TableMixin):
+    """
+    See the :ref:`Markers Plugin Documentation <markers-plugin>` for more details.
+
+    Only the following attributes and methods are available through the
+    :ref:`public plugin API <plugin-apis>`:
+
+    * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.show`
+    * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.open_in_tray`
+    * :meth:`clear_table`
+    * :meth:`~jdaviz.core.template_mixin.TableMixin.export_table`
+    """
+    template_file = __file__, "markers.vue"
+
+    @property
+    def user_api(self):
+        return PluginUserApi(self, expose=('clear_table', 'export_table',))
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self.config == 'cubeviz':
+            headers = ['spectral_axis', 'spectral_axis:unit',
+                       'slice', 'pixel',
+                       'world', 'value', 'value:unit', 'viewer']
+
+        elif self.config == 'imviz':
+            headers = ['pixel', 'pixel:unreliable',
+                       'world', 'world:unreliable',
+                       'value', 'value:unit', 'value:unreliable',
+                       'viewer']
+
+        elif self.config == 'specviz':
+            headers = ['spectral_axis', 'spectral_axis:unit',
+                       'index', 'value', 'value:unit']
+        elif self.config == 'specviz2d':
+            # TODO: add "index" if/when specviz2d supports plotting spectral_axis

```

Comment:
How do we cross-ref this TODO so that we won't forget to actually do it when it is time?